File: src/main/kotlin/com/lambdaschool/javasafezoos/SeedData.java

package com.lambdaschool.javasafezoos;

import com.lambdaschool.javasafezoos.models.Role;
import com.lambdaschool.javasafezoos.models.User;
import com.lambdaschool.javasafezoos.models.UserRoles;
import com.lambdaschool.javasafezoos.repositories.RoleRepository;
import com.lambdaschool.javasafezoos.repositories.UserRepository;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;

@Transactional
@Component
public class SeedData implements CommandLineRunner
{
    RoleRepository rolerepos;
    UserRepository userrepos;

    public SeedData(RoleRepository rolerepos, UserRepository userrepos)
    {
        this.rolerepos = rolerepos;
        this.userrepos = userrepos;
    }

    @Override
    public void run(String[] args) throws Exception
    {
        Role r1 = new Role("admin");
        Role r2 = new Role("zoodata");
        Role r3 = new Role("animaldata");
        Role r4 = new Role("mgr");

        ArrayList<UserRoles> roleAdminList = new ArrayList<>();
        roleAdminList.add(new UserRoles(new User(), r1));

        ArrayList<UserRoles> roleZooDataList = new ArrayList<>();
        roleZooDataList.add(new UserRoles(new User(), r2));

        ArrayList<UserRoles> roleAnimalDataList = new ArrayList<>();
        roleAnimalDataList.add(new UserRoles(new User(), r3));

        ArrayList<UserRoles> roleMgrList = new ArrayList<>();
        roleMgrList.add(new UserRoles(new User(), r4));

        rolerepos.save(r1);
        rolerepos.save(r2);
        rolerepos.save(r3);
        rolerepos.save(r4);

        User u1 = new User("useradmin", "1234", roleAdminList);
        User u2 = new User("userzoodata", "2345", roleZooDataList);
        User u3 = new User("useranimaldata", "3456", roleAnimalDataList);
        User u4 = new User("usermgr", "4567", roleMgrList);

        userrepos.save(u1);
        userrepos.save(u2);
        userrepos.save(u3);
        userrepos.save(u4);
    }
}

File: src/main/kotlin/com/lambdaschool/javasafezoos/SeedData.kt
package com.lambdaschool.javasafezoos

import com.lambdaschool.authenticatedusers.model.Quote
import com.lambdaschool.authenticatedusers.model.Role
import com.lambdaschool.authenticatedusers.model.User
import com.lambdaschool.authenticatedusers.model.UserRoles
import com.lambdaschool.authenticatedusers.repository.RoleRepository
import com.lambdaschool.authenticatedusers.repository.UserRepository
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

import java.util.ArrayList

@Transactional
@Component
class SeedData(internal var rolerepos: RoleRepository, internal var userrepos: UserRepository) : CommandLineRunner
{

    @Override
    @Throws(Exception::class)
    fun run(args: Array<String>)
    {
        val r1 = Role("admin")
        val r2 = Role("user")

        val admins = ArrayList()
        admins.add(UserRoles(User(), r1))
        admins.add(UserRoles(User(), r2))

        val users = ArrayList()
        users.add(UserRoles(User(), r2))

        rolerepos.save(r1)
        rolerepos.save(r2)
        val u1 = User("barnbarn", "ILuvM4th!", users)

        val u2 = User("admin", "password", admins)

        userrepos.save(u1)
        userrepos.save(u2)
    }
}
